ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -361,21 +361,10 @@ private void checkAndMaybeComplete() {
             new RouterException("DeleteOperation failed possibly because some replicas are unavailable",
                 RouterErrorCode.AmbryUnavailable));
       } else if (operationTracker.hasFailedOnNotFound()) {
-        /*
-        We are relying on the fact that at least one replica has the blob. Scenarios like below are rare, so that we don’t need to handle them for now.
-          1. put parallelism is 3 && all three replicas of originating dc are down && remote replication is not caught up
-          2. put parallelism is 2 && 2 replicas are down (note that 2 replicas down happen all the time) && replication has not caught up such that blob doesn’t exist in at least one other replica.
-              i.e, its very rare for 2 replicas to be down simultaneously and immediately after taking a POST (with parallelism 2)
-         */
-        if (operationTracker.getSuccessCount() > 0) {
-          routerMetrics.failedMaybeDueToUnavailableReplicasCount.inc();
-          operationException.set(
-              new RouterException("DeleteOperation failed possibly because of unavailable replicas",
-                  RouterErrorCode.AmbryUnavailable));
-        } else {
-          operationException.set(
-              new RouterException("DeleteOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
-        }
+        //TODO: Temporarily return 404 when delete operation returns two NOT_FOUND.
+        //TODO: Will check all local replicas and return succeed if one replica return succeed in this case.
+        operationException.set(
+            new RouterException("DeleteOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
```

Comment:
let's create a configuration for this and only turn it on for delivery cluster to return 503.

Reply:
What's the problem with returning 503 in this case? If for delete, the requirement of quorum is that there should be 2 successes, then why not let the user retry for eventual success?
Getting a 503 error from Ambry means that due to some replica unavailability, Ambry doesn't know what to do with the request, So please retry later.

Reply:
Delete is different from updateTTL. For updateTtl, when you get NOT_FOUND from a replica, you can wait until this replica replicating the PUT back. But for delete, replication won't solve the problem. For example. replica 1 has the PUT and DELETE, but replica 2 and replica 3 somehow had data wipeout in the past and they replicated from replica 1. However, when they replicate, the would see this blob id as deleted so they would total bypass this blob id. This would result in having no records for this blob id in replica 2 and 3 when replica 1 has the PUT and DELETE. This issue can't be solved waiting for more replication.